catalog/plugin/catalog-plugin-defaultsecurityattributevalues/src/main/java/org/codice/alliance/catalog/plugin/defaultsecurity/DefaultSecurityAttributeValuesPlugin.java

```diff
@@ -23,94 +25,104 @@
-    this.securityAttributes = securityAttributes;
-    this.subjectSupplier = subjectSupplier;
-
+      SystemHighAttributes systemHighAttributes,
+      Map<String, String> initialValues) {
+    this.securityAttributes = notNull(securityAttributes, "SecurityAttributes may not be null");
+    this.systemHighAttributes =
+        notNull(systemHighAttributes, "SystemHighAttributes may not be null");
     update(initialValues);
   }
 
-  public void update(Map<String, Object> properties) {
-    if (properties != null && !properties.isEmpty()) {
-      setClassification((String) properties.get(CLASSIFICATION_KEY));
-      setReleasability((String) properties.get(RELEASABILITY_KEY));
-      setCodewords((String) properties.get(CODEWORDS_KEY));
-      setDisseminationControls((String) properties.get(DISSEMINATION_CONTROLS_KEY));
-      setOtherDisseminationControls((String) properties.get(OTHER_DISSEMINATION_CONTROLS_KEY));
-      setOwnerProducer((String) properties.get(OWNER_PRODUCER_KEY));
-    }
+  /** Updates cached default system high attribute values. */
+  public void update(Map<String, String> properties) {
+    notEmpty(properties, "properties may not be empty");
+
+    final Map<String, String> newMetacardAttributeNameToConfigurationValueMap = new HashMap<>();
+
+    newMetacardAttributeNameToConfigurationValueMap.put(
+        Security.CLASSIFICATION, notEmpty(properties.get(CLASSIFICATION_CONFIGURATION_KEY)));
+    newMetacardAttributeNameToConfigurationValueMap.put(
+        Security.RELEASABILITY, notEmpty(properties.get(RELEASABILITY_CONFIGURATION_KEY)));
+    newMetacardAttributeNameToConfigurationValueMap.put(
+        Security.CODEWORDS, notEmpty(properties.get(CODEWORDS_CONFIGURATION_KEY)));
+    newMetacardAttributeNameToConfigurationValueMap.put(
+        Security.DISSEMINATION_CONTROLS,
+        notEmpty(properties.get(DISSEMINATION_CONTROLS_CONFIGURATION_KEY)));
+    newMetacardAttributeNameToConfigurationValueMap.put(
+        Security.OTHER_DISSEMINATION_CONTROLS,
+        notEmpty(properties.get(OTHER_DISSEMINATION_CONTROLS_CONFIGURATION_KEY)));
+    newMetacardAttributeNameToConfigurationValueMap.put(
+        Security.OWNER_PRODUCER, notEmpty(properties.get(OWNER_PRODUCER_CONFIGURATION_KEY)));
 
-    this.securityMarkings = getHighwaterSecurityMarkings();
+    metacardAttributeNameToSystemHighAttributeNameMap =
+        newMetacardAttributeNameToConfigurationValueMap;
   }
 
   /**
    * Adds default system high-water markings in the event that none of the policies were able to
    * apply security markings to this metacard.
    *
-   * @return Map of the metacard security markings to the value of their corresponding system high
-   *     attribute.
+   * @return metacard with possibly updated markings
    */
-  public Metacard addDefaults(Metacard metacard) {
-
-    if (securityMarkings
+  private Metacard addDefaults(final Metacard metacard) {
+    if (metacardAttributeNameToSystemHighAttributeNameMap
```

Comment:
✏️  Instead of inlining the `stream()` in the `if` conditions (here and on L139-142), should we consider doing what we did for the registry metacard check on L132 and extract `private` methods with descriptive names to make the conditions clearer?

Reply:
I extracted those 2 private methods.